### ISSUE TITLE
Using switch cases for log leveling leads to ts-error

### ISSUE THREAD

**Issue by JoinerDev** [xmldom/xmldom#754]
Hey,

in 0.9.x, it seems that the level "warn" was changed to "warning".
Whenever i try to process the validation levels with a switch like so, it throws a ts-error:

```
const parser = new DOMParser({
      onError: (level, msg, context) => {
        switch (level) {
          case 'warning':
            ...
            break
          case 'error':
            ...
            break
          case 'fatalError':
            ...
            break
          default:
            break
        }
      },
    })
```

The error reads: 
`Type '"warning"' is not comparable to type '"warn" | "error" | "fatalError"'.ts(2678)`

Workaround:
use `// @ts-expect-error`

The old one ("warn") does NOT throw a ts-error.
I'm assuming that somewhere you are still using "warn" instead of "warning" in your code? 
Maybe here?: 
https://github.com/xmldom/xmldom/blob/8279d8318bfef58e7ab63135f60ea2d417c6ccb4/test/get-test-parser.js#L5

xmldom version: 0.9.3
typescript version: 5.6.2
If you need further information, im happy to help!

**Comment by karfau** [comment]
I think the level/type should always have been `warning`, but I will double check once I have access to a real computer again.

**Comment by karfau** [comment]
fix is released as 0.9.4
